### DISCUSSION TITLE
Great News: All Bugs Will be Fixed In the Year 2041

### DISCUSSION THREAD

**Discussion by softworkz** [microsoft/microsoft-ui-xaml#9416] [General]
It's simple math:

- Bugs fixed in version 1.5: 24
- Bugs fixed in version 1.4: 16

Makes an average of 20. 
Releases are scheduled for about every 6 months - means 40 bugs get fixed in a year.

Current open bugs: 677

677 / 40 = 16.925 years

2024 + 17 = 2041

# YAY!

But only if everybody would stop reporting any new ones. If we take that into account:

In the past 12 months, 383 (now open) + 313 (now closed) bugs have been created (https://github.com/microsoft/microsoft-ui-xaml/issues?q=is%3Aissue++label%3Abug+sort%3Aupdated-desc++created%3A%3E2023-03-07)

Let's take only the open ones to extrapolate. Assuming that in the coming years, 383 bugs will be filed per year which can't be closed, then the count won't be zero, but instead

### 6,511 Bugs in 2041

# Oh....

**Comment by softworkz** [comment]
With 40 bugs getting fixed per year and about 40 work weeks per year, we can say that there's **one bug per week** that is getting fixed

This in turn raises the question: 

#### What is the one developer who is fixing the bugs,  asctually doing on the other 4 days each week?

**Reply by Fritierkelle** [reply]
Since I'm kind of struggling myself, can you tell me why you shouldn't switch to React, for example, or why you don't do it?
(I think MS Teams, for example, was programmed in React and is used a lot in Windows 11?! 
If the Windows Store is built with React, it has really crazy effects/possibilities.)

**Reply by softworkz** [reply]
Requirements are MS Store publishing for Xbox and using low-level media APIs, so it had to be a UWP app (due to Xbox) and later it turned out that UWP was too limiting for Desktop targets, which is why we did a WinUI3 version for that, which is largely the same as the UWP app but allows us to do a few things which aren't possible with UWP.
Large parts are happening in a WebView, though. For creating rich and complex UI in XAML, I wouldn't even consider using WinUI3 or UWP. Dev experience is inferior to the state of WPF from 10 years ago - not just in a few details but on almost all levels. 
It's the opposite of innovation and an embarrassing testimony for a company which has once been the undisputed leader in providing application development platforms and frameworks for LOB and other applications.
My criticism is not about dropping WinUI3 but about leaving users uninformed and letting them file proposals and bug reports, most of which will never be worked on.

**Reply by leowest77** [reply]
the new Windows Store was written in UWP afaik

**Comment by softworkz** [comment]
> What is the one developer who is fixing the bugs, asctually doing on the other 4 days each week?

I figured it out: Drawing ASCII art in code comments:



https://github.com/microsoft/microsoft-ui-xaml/blob/c8bd154c015b914e171a326481fef6532bc943de/dxaml/xcp/dxaml/lib/CueRenderer.cpp#L155-L168

**Comment by softworkz** [comment]
## Comparing with MAUI

When comparing with the fixes in  the MAUI changelogs from the past 12 months:

### WinUI3: 40 
### MAUI: 783

That's a factor of 20x

#### Details

| Version | Date | Fixes |
|--|--|--:|
8.0.7 | 14 Feb 24: |  70
8.0.6 |  17 Jan 24: |  61
8.0.3 |  Nov 14, 2023  | 56
8.0.100 |  Nov 7, 2023 |  13
8.0.0 |  Oct 31, 2023  | 29 
8.0.0-rc.2.9373  | Oct 10, 2023  | 48
7.0.96 |  Oct 10, 2023  | 26
8.0.0-rc.1.917 | 1 Sep 12, 2023 |  74
8.0.0-preview.7.8842  |  Aug 8, 2023  | 42
8.0.0-preview.6.8686 |  Jul 11, 2023  | 39
7.0.92 |  Jul 11, 2023  | 19
8.0.0-preview.5.8529  | Jun 13, 2023  | 49
8.0.0-preview.4.8333 |  May 16, 2023  | 69
6.0.553 Service Release |  11 Apr 26, 2023  | 4
7.0.86 Service Release |  6 May 9, 2023 |  3
8.0.0-preview.3.8149 |  Apr 11, 2023 |  79
7.0.81 Service Release |  4 Apr 11, 2023  | 35
8.0.0-preview.2.7871  | Mar 14, 2023 |  11
8.0.0-preview.1.7762  | Feb 21, 2023 |  56

**Reply by AndyDentFree** [reply]
Speaking as a former Xamarin Forms user recently considering MAUI (and sadly, rejecting for Flutter), I don't think comparing rate of bugs fixed is meaningful. MAUI shipped as a horrifically buggy mess so there's a lot more to clean up.

**Reply by MartyIX** [reply]
IMO MAUI got much better over recent months but there is still a lot of work to do. Perf is what bugs me.

**Reply by nikolayvpavlov** [reply]
MAUI works on top of WinUI 3, which makes the state of MAUI terribly pathetic.  I would not touch MAUI with a 10-meter pole.

**Comment by MartyIX** [comment]
Something to add to the stats: https://github.com/microsoft/microsoft-ui-xaml/compare/winui3/release/1.4-stable...winui3/release/1.5-stable ( Showing 55 changed files with 1,265 additions and 485 deletions.)

The comparison might be wrong because there is 
![image](https://github.com/microsoft/microsoft-ui-xaml/assets/203266/1be53805-1a08-4095-b385-57a1c311ca05)

**Comment by softworkz** [comment]
> The comparison might be wrong because there is

No, the comparison is correct, I had done a local file diff recently. The changes align with the release notes, that's why I didn't mention it. Given that this diff reflects the changes made over a period of 6 months, it confirms that just minimal work has been done from 1.4 to 1.5.

**Comment by Perksey** [comment]
This is an excellent step towards RFC 9225 compliance!

**Comment by vzhilong** [comment]
LOL, the bugs grown up to 1.7K now.......

**Comment by B1ue-Dev** [comment]
Grown up to 1.9K now.....

**Comment by ReginaldTang** [comment]
until theres new bugs
